compiler/rustc_llvm/llvm-wrapper/PassWrapper.cpp

```diff
@@ -945,12 +945,7 @@ LLVMRustOptimize(
         break;
       case LLVMRustOptStage::PreLinkThinLTO:
         MPM = PB.buildThinLTOPreLinkDefaultPipeline(OptLevel);
-        // The ThinLTOPreLink pipeline already includes ThinLTOBuffer passes. However, callback
-        // passes may still run afterwards. This means we need to run the buffer passes again.
-        // FIXME: In LLVM 13, the ThinLTOPreLink pipeline also runs OptimizerLastEPCallbacks
-        // before the RequiredLTOPreLinkPasses, in which case we can remove these hacks.
-        if (OptimizerLastEPCallbacks.empty())
-          NeedThinLTOBufferPasses = false;
+        NeedThinLTOBufferPasses = false;
```

Comment:
I believe you also need to drop the `OptStage != LLVMRustOptStage::PreLinkThinLTO` condition above and drop the manual OptimizerLastEPCallbacks scheduling below.

Reply:
Weird, locally `tests/ui/sanitize/new-llvm-pass-manager-thin-lto.rs` passes for me on `x86_64-pc-windows-msvc`.